sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1498,6 +1498,61 @@ def test_neighbors_validate_parameters(Estimator):
         nbrs.predict([[]])
 
 
+@pytest.mark.parametrize(
+    "Estimator",
+    [
+        neighbors.KNeighborsClassifier,
+        neighbors.RadiusNeighborsClassifier,
+        neighbors.KNeighborsRegressor,
+        neighbors.RadiusNeighborsRegressor,
+    ],
+)
+@pytest.mark.parametrize("n_features", [2, 100])
+def test_neighbors_minkowski_semimetric(Estimator, n_features):
```

Comment:
Can you add an assertion for a `ValueError` being raised when $p=0$? (Only the test should be modified, the parameters' validation via `_parameter_constraints` and friends will take care of raising the `ValueError`.)